app/controllers/users_controller.rb

```diff
@@ -1,23 +1,76 @@
 # Additional user methods in parallel with Devise -- all pertaining to call list
 class UsersController < ApplicationController
   before_action :retrieve_patients, only: [:add_patient, :remove_patient]
-  before_action :confirm_admin_user, only: [:new, :index]
+  before_action :confirm_admin_user, only: [:new, :index, :update]
+  before_action :find_user, only: [:update, :edit, :destroy]
+
   rescue_from Mongoid::Errors::DocumentNotFound, with: -> { head :bad_request }
 
   def index
     @users = User.all
   end
 
+  def edit
+  end
+
+  def search
+    if params[:search].empty?
+      @results = User.all
+    else
+      @results = User.search params[:search]
+    end
+    respond_to { |format| format.js }
+
+  end
+
+  def toggle_lock
+    # @user = User.find(params[:user_id])
+    # if @user == current_user
+    #   redirect_to edit_user_path @user
+    # else
+    #   if @user.access_locked?
+    #     flash[:notice] = 'Successfully unlocked ' + @user.email
+    #     @user.unlock_access!
+    #   else
+    #     flash[:notice] = 'Successfully locked ' + @user.email
+    #     @user.lock_access!
+    #   end
+    #   redirect_to edit_user_path @user
+    # end
+    # # TODO multiple locks?
+    # # respond_to { |format| format.js }
+  end
+
+  def reset_password
+    @user = User.find(params[:user_id])
+
+    # TODO doesn't work in dev
+    @user.send_reset_password_instructions
+
+    flash[:notice] = 'Successfully sent password reset instructions to ' + @user.email
```

Comment:
Can we use string interpolation here? `"blah blah blah #{@user.email}"`. More rails-y.